src/SingleProductDetailView.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from "react";
+import { Image } from "./Image.js";
+import "./SingleProductDetailView.css";
+
+export class SingleProductDetailView extends Component {
+  state = { product: {} };
+
+  componentDidMount() {
+    fetch("http://localhost:8000/listings/638754067")
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ product: data.results[0] });
+      })
+      .catch(error => console.error(error));
+  }
+
+  render() {
+    const { product } = this.state;
+
+    if (product.listing_id == null) {
+      return null;
+    }
+    function renderDescription() {
+      const description = product.description;
```

Comment:
do you need new `const`s for `description`, `title`, and `price` in these render functions? These functions should be able to just work with `product.description` etc